fv3core/stencils/del2cubed.py

```diff
@@ -73,47 +70,98 @@ def corner_fill(q: FloatField):
+        """
+        Perform hyperdiffusion damping/filtering
+
+        Args:
+            qdel (inout): Variable to be filterd
+            nmax: Number of times to apply filtering
+            cd: Damping coeffcient
+            km: Number of vertical levels (grid.npz)
+        """
+        ntimes = min(3, nmax)
+        for n in range(1, ntimes + 1):
+            nt = ntimes - n
+            origin = (self.grid.is_ - nt, self.grid.js - nt, 0)
+
+            # Fill in appropriate corner values
+            self._corner_fill(qdel)
+
+            if nt > 0:
+                corners.copy_corners_x_stencil(
+                    qdel,
+                    origin=(self.grid.isd, self.grid.jsd, 0),
+                    domain=(self.grid.nid, self.grid.njd, km),
+                )
+            nx = self.grid.nic + 2 * nt + 1
+            ny = self.grid.njc + 2 * nt
+            self._compute_zonal_flux(
+                self._fx, qdel, self.grid.del6_v, origin=origin, domain=(nx, ny, km)
+            )
 
-        if nt > 0:
-            corners.copy_corners_x_stencil(
-                qdel, origin=(grid.isd, grid.jsd, 0), domain=(grid.nid, grid.njd, km)
+            if nt > 0:
+                corners.copy_corners_y_stencil(
+                    qdel,
+                    origin=(self.grid.isd, self.grid.jsd, 0),
+                    domain=(self.grid.nid, self.grid.njd, km),
+                )
+            nx = self.grid.nic + 2 * nt
+            ny = self.grid.njc + 2 * nt + 1
+            self._compute_meridional_flux(
+                self._fy, qdel, self.grid.del6_u, origin=origin, domain=(nx, ny, km)
             )
-        nx = grid.nic + 2 * nt + 1  # (grid.ie+nt+1) - (grid.is_-nt) + 1
-        ny = grid.njc + 2 * nt  # (grid.je+nt) - (grid.js-nt) + 1
-        compute_zonal_flux(fx, qdel, grid.del6_v, origin=origin, domain=(nx, ny, km))
 
-        if nt > 0:
-            corners.copy_corners_y_stencil(
-                qdel, origin=(grid.isd, grid.jsd, 0), domain=(grid.nid, grid.njd, km)
+            # Update q values
+            ny = self.grid.njc + 2 * nt
+            self._update_q(
+                qdel,
+                self.grid.rarea,
+                self._fx,
+                self._fy,
+                cd,
+                origin=origin,
+                domain=(nx, ny, km),
             )
-        nx = grid.nic + 2 * nt  # (grid.ie+nt) - (grid.is_-nt) + 1
-        ny = grid.njc + 2 * nt + 1  # (grid.je+nt+1) - (grid.js-nt) + 1
-        compute_meridional_flux(
-            fy, qdel, grid.del6_u, origin=origin, domain=(nx, ny, km)
-        )
 
-        # Update q values
-        ny = grid.njc + 2 * nt  # (grid.je+nt) - (grid.js-nt) + 1
-        update_q(qdel, grid.rarea, fx, fy, cd, origin=origin, domain=(nx, ny, km))
+
+def compute(qdel: FloatField, nmax: int, cd: float, km: int):
+    hyperdiffusion = utils.cached_stencil_class(HyperdiffusionDamping)(
+        spec.grid, qdel, km
+    )
+    hyperdiffusion(qdel, nmax, cd, km)
```

Comment:
ah very interesting choice to put this in the compute method! This way you don't have to change the test translate code and other code that calls it. What we have been doing with the other ones is remove the compute method alltogether and then initialize the new class where it gets called. The places this gets called, dyn_core and fv_dynamics, are already model objects, and so this initialization of HyperdiffusionDamping can be done in each of their __init__ methods, and we no longer need to cache the object in the runtime explicitly. If you do this, though you will also need to initialize HyperdiffusionDamping in tests/translate/translate_del2cubed.py

Reply:
Thanks for pointing this out! HyperdiffusionDamping is now initialized in both dyn_core and fv_dynamics.